libraries/extensions/ros2-bridge/python/src/typed/serialize.rs

```diff
@@ -31,81 +34,154 @@ impl serde::Serialize for TypedValue<'_> {
+                } else {
+                    let mut s = serializer.serialize_seq(Some(array.len()))?;
+                    for value in array.iter() {
+                        s.serialize_element(&value)?;
+                    }
+                    s.end()
+                }
             }
             DataType::Float64 => {
-                let int_array: Float64Array = self.value.clone().into();
-                let number = int_array.value(0);
-                serializer.serialize_f64(number)
+                let array: &Float64Array = self.value.as_primitive();
+                if array.len() == 1 {
+                    let number = array.value(0);
+                    serializer.serialize_f64(number)
+                } else {
+                    let mut s = serializer.serialize_seq(Some(array.len()))?;
+                    for value in array.iter() {
+                        s.serialize_element(&value)?;
+                    }
+                    s.end()
+                }
             }
             DataType::Utf8 => {
-                let int_array: StringArray = self.value.clone().into();
+                let int_array: &StringArray = self.value.as_string();
                 let string = int_array.value(0);
                 serializer.serialize_str(string)
             }
             DataType::List(field) => {
-                let list_array: ListArray = self.value.clone().into();
-                let values = list_array.values();
-                let mut s = serializer.serialize_seq(Some(values.len()))?;
+                let list_array: &ListArray = self.value.as_list();
+                // let values: &UInt16Array = list_array.values().as_primitive();
+                let mut s = serializer.serialize_seq(Some(list_array.len()))?;
                 for value in list_array.iter() {
-                    let value = match value {
-                        Some(value) => value.to_data(),
-                        None => {
-                            return Err(serde::ser::Error::custom(
-                                "Value in ListArray is null and not yet supported".to_string(),
-                            ))
-                        }
-                    };
-
-                    s.serialize_element(&TypedValue {
-                        value: &value,
-                        type_info: &TypeInfo {
-                            data_type: field.data_type().clone(),
-                            defaults: self.type_info.defaults.clone(),
-                        },
-                    })?;
+                    if let Some(value) = value {
+                        s.serialize_element(&TypedValue {
+                            value: &value,
+                            type_info: &TypeInfo {
+                                data_type: field.data_type().clone(),
+                                defaults: self.type_info.defaults.clone(),
+                            },
+                        })?;
+                    }
```

Comment:
I'm not sure whether ignoring `null` values is a good idea. This would reorder the list elements and make it shorter....